context/user_plane_information.go

```diff
@@ -354,13 +355,73 @@ func (upi *UserPlaneInformation) InsertSmfUserPlaneNode(name string, node *facto
 	return nil
 }
 
-// update UPF info
-func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, node *factory.UPNode) error {
-	//What all to update- nssai/dnn info
-	logger.UPNodeLog.Infof("UPNode[%v] to update, content[%v]\n", name, node)
-	logger.UPNodeLog.Debugf("content of map[UPNodes] %v \n", upi.UPNodes)
-	upi.DeleteSmfUserPlaneNode(name, node)
-	upi.InsertSmfUserPlaneNode(name, node)
+func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, newNode *factory.UPNode) error {
+	logger.UPNodeLog.Infof("UPNode [%v] to update, content[%v]\n", name, newNode)
+
+	existingNode, exists := upi.UPNodes[name]
+	if !exists {
+		return fmt.Errorf("UPNode [%v] does not exist", name)
+	}
+
+	existingNode.Port = newNode.Port
+
+	if existingNode.Type == UPNODE_AN {
+		existingNode.ANIP = net.ParseIP(newNode.ANIP)
+		return nil
```

Comment:
Do we really want to return here? I see some code towards the end of function that looks general, like a logging call and some assignments.

Reply:
Done, we're now following the same structure as the insert func